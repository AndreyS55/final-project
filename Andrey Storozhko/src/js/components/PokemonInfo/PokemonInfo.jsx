import React from 'react';
import { connect } from 'react-redux';
import PokemonImage from "../PokemonImage/PokemonImage";
import { fetchSinglePokemon, unmountComponentSingle } from '../../actions/singlePokemonActions';

import styles from './PokemonInfo.scss';

class PokemonInfo extends React.Component {
  // componentWillMount() {
  //   this.props.fetchSinglePokemon(this.props.id)
  // }

  componentWillUnmount() {
    this.props.unmountComponentSingle()
  }

  render() {
    console.log(this.props.pokemon);
    return (
      <div className={styles.pokemonInfo}>
        <div className={styles.poka}>
          <PokemonImage id={this.props.pokemon.id}/>
          <div>ID: {this.props.pokemon.id}</div>
          <div>NAME: {this.props.pokemon.name}</div>
          {/*{this.props.pokemon.catched.length ? <div>STATUS: CATCHED</div> : <div>STATUS: NOT CATCHED</div>}
          {this.props.pokemon.catched.length ? <div>CATCH DATE: {this.props.pokemon.catched[0].date.toLocaleString()}</div> : null}*/}
          <button onClick={this.props.handleOpen}>CLOSE</button>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    pokemon: state.singlePokemon.pokemon,
    error: state.singlePokemon.error,
    isLoading: state.singlePokemon.isLoading
  };
};

const mapDispatchToProps = (dispatch) => ({
  fetchSinglePokemon: (id) => dispatch(fetchSinglePokemon(id)),
  unmountComponentSingle: () => dispatch(unmountComponentSingle())
});

export default connect(mapStateToProps, mapDispatchToProps)(PokemonInfo);
