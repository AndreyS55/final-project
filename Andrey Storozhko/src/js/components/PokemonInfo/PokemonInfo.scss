@import '../../../styles/environment/mixins';
@import '../../../styles/environment/colors';

.pokemonInfo {
  display: flex;
  justify-content: center;
  align-items: center;
  position: fixed;
  z-index: 999;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  background-color: rgba(95, 95, 95, 0.8);
}

.pokemon {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 500px;
  width: 800px;
  background: fixed url("../../../img/cardBackground.png") center no-repeat;
  border-radius: 30px;

  @include tablets() {
    width: 90%;
    height: 90%;
    border-radius: 0;
    flex-direction: column;
    justify-content: start;
    background: fixed url("../../../img/cardBackgroundTablet.png") right no-repeat;
  }

  @include phones() {
    width: 100%;
    height: 100%;
    border-radius: 0;
    flex-direction: column;
    justify-content: start;
    background: fixed url("../../../img/cardBackgroundTablet.png") right no-repeat;
  }
}

.singlePokemonImage {
  max-width: 40%;

  @include tablets() {
    max-width: 80%;
  }

  @include phones() {
    max-width: 80%;
    margin-top: 20%;
  }
}

.infoText {
  font-family: 'Permanent Marker', cursive;
  font-size: 1.8rem;
  align-self: start;
  padding: 20px;
  margin: 60px 40px 0 0;
  border-radius: 20px;
  background-color: $wht;

  @include desktop() {
    font-size: 1.2rem;
  }

  @include laptop() {
    font-size: 1.5rem;
  }

  @include tablets() {
    font-size: 1.8rem;
    margin: 0;
    align-self: center;
  }

  @include phones() {
    font-size: 1.2rem;
    margin: 0;
  }
}

.infoHead {
  font-weight: bold;
}

.pokemonCloseButton {
  position: absolute;
  top: 30px;
  right: 30px;
  border: none;
  background: none;
  font-size: 1.5rem;
  cursor: pointer;
  transform: translate(50%, -50%);

  &:hover {
    font-size: 2rem;
  }
}