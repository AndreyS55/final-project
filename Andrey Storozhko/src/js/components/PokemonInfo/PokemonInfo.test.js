import React from 'react';
import PokemonInfo from './PokemonInfo';

it('should render a PokemonInfo', () => {
  const wrapper = shallow(<PokemonInfo pokemon={
    {
      name: 'bulbasaur',
      id: 1,
      caught: [
        {
          name: 'bulbasaur',
          date: '08.02.2019',
          pokemonId: 1,
          id: 1
        }
      ]
    }
  }
  />);
  expect(wrapper).toMatchSnapshot();
});
