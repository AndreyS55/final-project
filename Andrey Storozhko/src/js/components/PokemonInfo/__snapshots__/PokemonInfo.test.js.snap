// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a PokemonInfo 1`] = `
<div
  className="pokemonInfo"
>
  <div
    className="pokemon"
  >
    <figure
      className="singlePokemonImage"
    >
      <PokemonImage
        id={1}
      />
    </figure>
    <div
      className="infoText"
    >
      <div>
        <span
          className="infoHead"
        >
          ID: 
        </span>
        1
      </div>
      <div>
        <span
          className="infoHead"
        >
          NAME: 
        </span>
        BULBASAUR
      </div>
      <div>
        <span
          className="infoHead"
        >
          STATUS: 
        </span>
        CAUGHT
      </div>
      <div>
        <span
          className="infoHead"
        >
          CATCH DATE: 
        </span>
        08.02.2019
      </div>
    </div>
    <button
      className="pokemonCloseButton"
      onClick={[Function]}
    >
      ‚ùå
    </button>
  </div>
</div>
`;
